bin/lib/amazon.py

```diff
@@ -17,6 +17,10 @@ def __getattr__(self, attr):
         self.__ensure_setup()
         return getattr(self.__obj, attr)
 
+# this is a free function to avoid potentially shadowing any underlying members
```

Comment:
Thanks for the comment...though I don't quite understand (low priority, perhaps ping me on slack?)

Reply:
The motivating example is where your other comment is below - I need to forcefully initialize boto _once_, and in some of my testing I observed weird exceptions and whatnot when it happened as part of the DeploymentJob.

Its probably not an issue anymore, since now the deploy job runs as part of a command which will initialize boto as part of other operations, but I left it to be safe.

The comment about being a free function is as opposed to making it a member of `LazyObjectWrapper`, where it could potentially shadow a name from an underlying type we try to wrap.

EX:
```python
class Example(object):
  def init(self):
    print("Hello")

class LazyObjectWrapper(object):
    def __init__(self, fn):
        self.__fn = fn
        self.__setup = False
        self.__obj = None

    def __ensure_setup(self):
        if not self.__setup:
            self.__obj = self.__fn()
            self.__setup = True

    # instead of the free function
    def init(self):
        self.__ensure_setup()

    def __getattr__(self, attr):
        self.__ensure_setup()
        return getattr(self.__obj, attr)

ex = LazyObjectWrapper(lambda: Example())
ex.init() # will call LazyObjectWrapper.init
ex.init() # will still call LazyObjectWrapper.init
```